aleemer/Internify
# [IN-69] Create User and POST to back-end when user logins

#### User Story
[[IN-69] Nice](https://chaosneutral.atlassian.net/jira/software/projects/IN/boards/1?selectedIssue=IN-69)

#### Changes made
TLDR: Logging in from either `Landing` or `Login` page makes call to back-end
- Cleaned up `Profile` page to reference data from state object
- Updated `UserSchema` to include `authId`
- Logging in with Google from the `Landing` page calls `POST` request to the back-end to make a new `User` object and uses the result from the request to update the redux state with the new `User` object
- Logging in with Google from the `SignIn` page calls `GET` request to the back-end to retrieve the `User` object from the db and uses the result from the request to update the redux state
- Logging in with email from the `Landing` page calls `POST` request to the back-end to make a new `User` object and uses the result from the request to update the redux state with the new `User` object
- Logging in with email from the `SignIn` page calls `GET` request to the back-end to retrieve the `User` object from the db and uses the result from the request to update the redux state
- **Fixed GET User reducers and back-end resolvers**
- **Fixed POST User reducers and back-end resolvers**

#### Does your new code introduce new warnings on dev console?
No

#### Test Steps
1. Log in with Google from the `Landing` page
2. Open the dev console and check the redux tab, you should see the current user updated in the user state
3. Open MongoDB Compass (or any other MongoDB UI) and connect to our db
4. You should see the new `User` object in our `userdatas` collection
5. Sign out
6. Log in with Google from the `Login` page
7. Open the dev console and check the redux tab, you should see the current user updated in the user state
8. Repeat steps 1 to 7 with sign in with email
